plugins/cad/src/components/RepositoryPage/components/AdvancedRepositoryOptions.tsx

```diff
@@ -41,10 +48,33 @@ export const AdvancedRepositoryOptions = () => {
   };
 
   const executeUnregisterRepository = async (): Promise<void> => {
-    await api.unregisterRepository(repositoryName);
+    const repoSecretName =
+      repositorySummary.repository.spec.git?.secretRef?.name;
+    const repoNamespace = 
+      repositorySummary.repository.metadata.namespace;
+
+    if(repoSecretName && !checkUsedByOtherRepository(repoSecretName)) {
```

Comment:
`checkUsedByOtherRepository` is an asynchronous function so we'll need to await the response.